### PR TITLE
Feature/sg 468 detection transform support for any number of channels

### PR DESCRIPTION
**Pull request by shaydeci**
- Modified Mixup and HSV to work with any number of channels.
- Warning is displayed when applying HSV with additional channels.
- Added an integration test, that checks the support of our coco base recipe transforms (all detection transforms) compatibility with 6 channels.

### CODE REVIEW

File: src/super_gradients/training/transforms/transforms.py [hunk 1]
```diff
@@ -567,7 +582,7 @@ def __call__(self, sample: dict):
             jit_factor = random.uniform(*self.mixup_scale)
 
             if len(img.shape) == 3:
-                cp_img = np.ones((self.input_dim[0], self.input_dim[1], 3), dtype=np.uint8) * 114
+                cp_img = np.ones((self.input_dim[0], self.input_dim[1], img.shape[2]), dtype=np.uint8) * 114
```
Comment (ofrimasad) [inline src/super_gradients/training/transforms/transforms.py:585]: if you added border_value, why is 114 hard codded here?

Comment (ofrimasad) [reply src/super_gradients/training/transforms/transforms.py:585]: what if if len(img.shape) == 2? (grayscale) I see you are checking that in line 607 but not here

File: src/super_gradients/training/transforms/transforms.py [hunk 2]
```diff
@@ -699,8 +728,18 @@ def __init__(self, prob: float, hgain: float = 0.5, sgain: float = 0.5, vgain: f
         self.sgain = sgain
         self.vgain = vgain
         self.bgr_channels = bgr_channels
+        self._additional_channels_warned = False
 
     def __call__(self, sample: dict) -> dict:
+        if sample["image"].shape[2] > 3 and not self._additional_channels_warned:
```
Comment (ofrimasad) [inline src/super_gradients/training/transforms/transforms.py:734]: what if if len(img.shape) == 2? (grayscale) this will crash

**Review by ofrimasad — COMMENTED** [review]
a few comments inline

File: src/super_gradients/training/transforms/transforms.py (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/training/transforms/transforms.py]: fixed- raised exception when channels < 3 in latest.

Comment (shaydeci) [reply src/super_gradients/training/transforms/transforms.py]: Also added border value to all transforms where 114 was hard coded.

**Review by ofrimasad — APPROVED** [review]
LGTM

**Pull request merged**